scripts/remote_deployment/deploy_calibration.sh

```diff
@@ -0,0 +1,214 @@
+#!/bin/bash
+
+# IPC Quick Start Script
+# See also https://github.com/consensus-shipyard/ipc/blob/main/docs/ipc/quickstart-calibration.md
+
+# Known issues:
+# 1. Need to previously manual enable sudo without password on the host
+# 2. You may need to rerun the script after docker installation for the first time
+
+# Exit on any error
+set -e
+
+# Print commands as we execute
+# set -x
+
+PREFIX='------'
+IPC_FOLDER=${HOME}/ipc
+IPC_CLI=${IPC_FOLDER}/target/release/ipc-cli
+IPC_CONFIG_FOLDER=${HOME}/.ipc
+
+wallet_addresses=()
+address_pubkeys=()
+CMT_P2P_HOST_PORTS=(26656 26756 26856)
+CMT_RPC_HOST_PORTS=(26657 26757 26857)
+ETHAPI_HOST_PORTS=(8545 8645 8745)
+RESOLVER_HOST_PORTS=(26655 26755 26855)
+
+# Step 1: Prepare system for building and running IPC
+
+# Step 1.1: Install build dependencies
+echo "${PREFIX} Installing build dependencies..."
+sudo apt update && sudo apt install build-essential libssl-dev mesa-opencl-icd ocl-icd-opencl-dev gcc git bzr jq pkg-config curl clang hwloc libhwloc-dev wget ca-certificates gnupg -y
+
+# Step 1.2: Install rust + cargo
+echo "$PREFIX Check rustc & cargo..."
+if which cargo ; then
+  echo "$PREFIX rustc & cargo already installed."
+else
+  echo "$PREFIX Need to install rustc & cargo"
+  curl https://sh.rustup.rs -sSf | sh -s -- -y
+  # Refresh env
+  source ${HOME}/.bashrc
+fi
+
+# Step 1.3: Install cargo-make and toml-cli
+# Install cargo make
+cargo install cargo-make
+# Install toml-cli
+cargo install toml-cli
+
+# Step 1.3: Install Foundry
+echo "$PREFIX Check foundry..."
+if which foundryup ; then
+  echo "$PREFIX foundry is already installed."
+else
+  echo "$PREFIX Need to install foundry"
+  curl -L https://foundry.paradigm.xyz | bash
+  foundryup
+fi
+
+# Step 1.4: Install docker
+echo "$PREFIX check docker"
+if which docker ; then
+  echo "$PREFIX docker is already installed."
+else
+  echo "$PREFIX Need to install docker"
+  # Add Docker's official GPG key:
+  sudo apt-get update
```

Comment:
Do we have `sudo` on CI? Can we do `sudo deploy_calibration.sh` instead of putting it inside the script?

Reply:
I think we will not run this script in the CI server, but on the dedicated host. So we should have sudo there.